secret_store/src/http_listener.rs

```diff
@@ -93,6 +93,13 @@ impl<T> KeyServer for KeyServerHttpListener<T> where T: KeyServer + 'static {
 	}
 }
 
+impl<T> Drop for KeyServerHttpListener<T> where T: KeyServer + 'static {
+	fn drop(&mut self) {
+		// ignore error as we are dropping anyway
+		let _ = self._http_server.close();
```

Comment:
Nope. That was the rason behind "Fixed closing Parity by ^C when secretstore was working" issue. You must call `close()` explicitely